src/sentry/static/sentry/app/stores/discoverSavedQueriesStore.tsx

```diff
@@ -1,8 +1,29 @@
 import Reflux from 'reflux';
-
-import {SavedQuery} from 'app/views/discover/types';
 import DiscoverSavedQueryActions from 'app/actions/discoverSavedQueryActions';
 
+type Versions = 1 | 2;
+
+export type NewQuery = {
+  id: string | undefined;
+  version: Versions;
+  name: string;
+  projects: number[];
+  fields: string[];
+  fieldnames: string[];
+  query: string;
+  orderby?: string;
+  range?: string;
+  start?: string;
+  end?: string;
+};
+
+export type SavedQuery = NewQuery & {
+  id: string;
+  dateCreated: string;
+  dateUpdated: string;
+  createdBy?: string;
```

Comment:
When does this happen? Default/system queries?

Reply:
This happens *after* the query is saved by the API endpoint.